### PR TITLE
New duplicate matches support UI

### PR DESCRIPTION
**Pull request by tomas-stefano**
## Context

This adds the new duplicate matches UI to the support interface only when the feature duplicate matching is enabled.

#### Before (listing duplicate matches)

Screenshot 2022-01-11 at 13 11 25

#### After (listing duplicate matches)

Screenshot 2022-01-11 at 13 02 14

#### Before (show duplicate match)

Screenshot 2022-01-11 at 13 11 54

#### After (show duplicate match)

Screenshot 2022-01-11 at 13 02 01

## Guidance to review

- [ ] Read commits
- [ ] Read the trello card
- [ ] Run locally, enable duplicate matching feature flag and enter the support -> candidates -> duplicate candidate matches tab

## Link to Trello card

https://trello.com/c/dpHoZyRQ/4250-create-new-duplicate-matches-ui

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

## Out of the scope of this PR

The following feature will be done in a separate card and separate PR:

-  Pagination
-  Block/unblock submissions
-  Lock/unlock accounts

### CODE REVIEW

File: app/components/support_interface/duplicate_matches_candidates_table_component.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
+module SupportInterface
+  class DuplicateMatchesCandidatesTableComponent < ViewComponent::Base
+    include ViewHelper
+
+    DuplicatedMatch = Struct.new(
+      :email_address, :candidate_id, :created_at, :name, :date_of_birth, :address, :application_status, :account_status,
+      keyword_init: true
```
Comment (duncanjbrown) [inline app/components/support_interface/duplicate_matches_candidates_table_component.rb:7]: Even better — TIL!

File: app/components/support_interface/duplicate_matches_table_component.html.erb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      <th scope="col" class="govuk-table__header">Match</th>
+      <th scope="col" class="govuk-table__header">Date</th>
+    </tr>
+  </thead>
+
+  <tbody class="govuk-table__body">
+    <% @matches.each do |match| %>
+      <tr class="govuk-table__row">
+        <td class="govuk-table__cell">
+          <%= govuk_link_to SupportInterface::DuplicateMatchesTableComponent.description_for(match), support_interface_duplicate_match_path(match) %>
```
Comment (duncanjbrown) [inline app/components/support_interface/duplicate_matches_table_component.html.erb:13]: I think elsewhere we make helpers like this instance methods and call without the qualifying class?

File: app/views/support_interface/duplicate_matches/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,33 @@
+<%= render 'support_interface/candidates/candidates_navigation', title: 'Duplicate candidate matches' %>
+
+<nav class="govuk-body">
+  <% if params[:resolved] == 'true' %>
```
Comment (duncanjbrown) [inline app/views/support_interface/duplicate_matches/index.html.erb:4]: Maybe assign this in the controller, now that we're using it ~twice~ three times? (once here, once in the query, once (without checking the value) below)

File: app/components/support_interface/duplicate_matches_table_component.html.erb (hunk 2 above)
Comment (duncanjbrown) [reply app/components/support_interface/duplicate_matches_table_component.html.erb:13]: ah — I see, it's called out of context below. carry on!

File: config/initializers/date_formats.rb [hunk 4]
```diff
@@ -1,6 +1,7 @@
 Time::DATE_FORMATS[:govuk_date] = '%-d %B %Y'
 Time::DATE_FORMATS[:govuk_date_short_month] = '%-d %b %Y'
 Date::DATE_FORMATS[:govuk_date] = '%-d %B %Y'
+Date::DATE_FORMATS[:govuk_date_short_month] = '%-d %b %Y'
```
Comment (duncanjbrown) [inline config/initializers/date_formats.rb:4]: I think duplicated from 2 lines up?

**Review by duncanjbrown — COMMENTED** [review]
This looks excellent! nit about `params[:resolved]` inline. will test through the front end first thing in the morning

File: config/initializers/date_formats.rb (hunk 4 above)
Comment (tomas-stefano) [reply config/initializers/date_formats.rb:4]: I had the same thought when I read the first time. But actually, one is for Date and another for Time.

File: app/views/support_interface/duplicate_matches/index.html.erb (hunk 3 above)
Comment (tomas-stefano) [reply app/views/support_interface/duplicate_matches/index.html.erb]: Done ♻️

**Review by duncanjbrown — APPROVED** [review]

**Pull request merged**